packages/driver/src/cy/commands/sessions/index.ts

```diff
@@ -170,11 +170,23 @@ export default function (Commands, Cypress, cy) {
           displayName: recreateSession ? 'Recreate session' : 'Create new session',
           message: '',
           type: 'system',
-        }, () => {
+        }, (setupLogGroup) => {
           return cy.then(async () => {
+            // Catch when a cypress command fails in the setup function to correctly update log status
+            // before failing command and ending command queue.
+            cy.state('onCommandFailed', (err) => {
```

Comment:
does 'onCommandFailed' have to live in state? It would be great if we didn't have to rely on global (to cy) state for this.

Reply:
I spent (quite a bit of) time looking into what alternative options we could implement here. Unfortunately, with the current command queue structure, I think this is the quickest win. To work around this would require some reworking on how the command queue is setup for nested-commands. (also sessions is just the weirdo command that is essentially a "workflow" what requires a series of commands before the workflow command is successful).